torchrl/envs/transforms/transforms.py

```diff
@@ -3017,6 +3026,7 @@ def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
                 dtype=torch.int64,
                 device=tensordict.device,
             )
+        _reset = _reset.view(tensordict.batch_size, -1).any(-1)
         step_count[_reset] = 0
```

Comment:
This way we are only counting steps within the batch_size.
This is cool but someone may want to count steps on the done.shape

Reply:
So currently, with that line, we are reducing the _reset shape to the batch_size. 
This is because the step_count counts steps in the batch_size and not in the don_spec.shape

But the _reset shape could be bigger (as you could have an agent dimension in the done_spec)
In the current scenario, thus, step_count would only count in the batch_size (and not for each agent for example)

If we make step_count follow the done_spec as well, then we can count also per agent steps

Reply:
so essentially
```suggestion
        step_count[_reset.view_as(step_count)] = 0
```
instead of 
```
        _reset = _reset.view(*tensordict.batch_size, -1).any(-1)
        step_count[_reset] = 0
```
right?

Reply:
Mmh no, that wouldn't work because _reset would need a restuction operation like sum.

What i am proposing is to replace 
```python
  if step_count is None:
          step_count = torch.zeros(
              tensordict.batch_size,
              dtype=torch.int64,
              device=tensordict.device,
          )
```
with 
```python
  if step_count is None:
          step_count = torch.zeros(
              self.parent.done_spec.batch_size,
              dtype=torch.int64,
              device=tensordict.device,
          )
```
so then we can just do 
```python
step_count[_reset] = 0
```

Reply:
lgtm, the only thing that I see is that we'd have a step_count that now has a trailing singleton dim that we currently don't have. This may be a problem down the line.
I'm open to the idea of removing the trailing singleton for done and reward, but before we do that I would not introduce new ones in the midst

Reply:
It would have the trailing singleton as "is_init", "done" and "_reset" have.  In my view, these 3 keys and more like "step_count", "truncated", etc. should all follow done_spec. Whatever done_spec is (now it happens to have a trailing singleton).

But more than the trailing singleton my point is about multi-agent.

Consider the example where we have 
```python
td.batch_size = (32,64)
td["done"].shape = (32,64, n_agents, 1)
```
then step_count would be
```python
td["step_count"].shape = (32,64)
```
while it  better to have
```python
td["step_count"].shape = (32,64,n_agents,1)
```

Reply:
the thing is not to assume that done_spec.shape is always `(*batch_size,1)`. If we make this assumption in the codebase, we bring problems down the line

Reply:
I'll make the modifications to step_count and notify when this pr is ready for review